src/main/java/com/acrolinx/proxy/AcrolinxProxyHttpServlet.java

```diff
@@ -94,6 +95,42 @@ private static void setRequestHeader(
     httpRequestBase.setHeader(headerName, headerValue);
   }
 
+  private static void transferResponseBodyWitAdditionalHeaders(
+      HttpServletResponse httpServletResponse, HttpEntity httpEntity) throws IOException {
+    try (InputStream inputStream = httpEntity.getContent();
+        OutputStream outputStream = httpServletResponse.getOutputStream()) {
+      Header contentTypeHeader = httpEntity.getContentType();
+
+      if (contentTypeHeader != null) {
+        httpServletResponse.setContentType(contentTypeHeader.getValue());
+      }
+
+      try (ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream()) {
+        long numberOfTransferredBytes = inputStream.transferTo(byteArrayOutputStream);
+        httpServletResponse.setContentLength((int) numberOfTransferredBytes);
+
+        // all response headers must be set before writing to the OutputStream
+        outputStream.write(byteArrayOutputStream.toByteArray());
+      }
+
+      logger.debug("Forwarded response to client");
+    }
+  }
+
+  private static void transferResponseHeaders(
+      HttpServletResponse httpServletResponse, CloseableHttpResponse httpResponse) {
```

Comment:
could we change the second parameter type to HttpResponse? Then type and name are also aligned.